docs/doxygen/migration/mqtt.txt

```diff
@@ -0,0 +1,127 @@
+  <b>Workaround:</b> When a version 202009.00 API call does not return @ref MQTTSuccess, the application should re-establish
+  the underlying transport connection and then call @ref MQTT_Connect. The application should employ an exponential backoff strategy when re-establishing connections.
+- <u>Thread support</u> <br>
+  Version 3 has an `_ENABLE_THREAD_SUPPORT_` macro for multi-threading use cases. Version 202009.00 has been redesigned to use only a single execution thread, and is
+  not thread-safe.
+
+@section migration_mqtt_datatypes Data Types
+@brief The following table lists equivalent data types in v3 and 202009.00.
+
+<table>
+    <tr>
+        <th>Version 3</th>
+        <th>Version 202009.00</th>
+        <th>Notes</th>
+    </tr>
+    <tr>
+        <td>enum QoS</td>
+        <td>@ref MQTTQoS_t</td>
+        <td>MQTT Quality of service.</td>
+    </tr>
+    <tr>
+        <td>AWS_IoT_Client</td>
+        <td>@ref MQTTContext_t</td>
+        <td>MQTT connection handle.</td>
+    </tr>
+    <tr>
+        <td>IoT_Publish_Message_Params</td>
+        <td>@ref MQTTPublishInfo_t</td>
+        <td>Parameters of an MQTT publish.</td>
+    </tr>
+    <tr>
+        <td>IoT_MQTT_Will_Options</td>
+        <td>@ref MQTTPublishInfo_t</td>
+        <td>Optional Will parameter of an MQTT connect. </td>
+    </tr>
+    <tr>
+        <td>IoT_Client_Connect_Params</td>
+        <td>@ref MQTTConnectInfo_t</td>
+        <td>Parameter of an MQTT connect. </td>
+    </tr>
+    <tr>
+        <td>IoT_Client_Init_Params</td>
+        <td><i>None</i></td>
+        <td>The members of this struct in v3 handled setup of the network connection.<br>This is handled outside of the MQTT library in 202009.00.</td>
+    </tr>
+    <tr>
+        <td>pApplicationHandler_t</td>
+        <td>@ref MQTTEventCallback_t</td>
+        <td>Function pointer of application callback to receive incoming data.</td>
+    </tr>
+</table>
+
+@section migration_mqtt_functions Functions
+@brief The following table lists equivalent API functions in v3 and 202009.00. These functions are the API functions declared in:
+- In v3: <a href="https://github.com/aws/aws-iot-device-sdk-embedded-C/blob/0da87f06f1501fad7b03b5d89a1322cc8a8848b4/include/aws_iot_mqtt_client_interface.h">aws_iot_mqtt_client_interface.h</a>
+- In 202009.00: [MQTT functions](@ref mqtt_functions)
+
+<table>
+    <tr>
+        <th>Version 3</th>
+        <th>Version 202009.00</th>
+        <th>Notes</th>
+    </tr>
+    <tr>
+        <td>aws_iot_mqtt_init</td>
+        <td>@ref MQTT_Init</td>
+        <td></td>
+    </tr>
+    <tr>
+        <td>aws_iot_mqtt_free</td>
+        <td><i>None</i></td>
+        <td>202009.00 has nothing to free, as it does not use synchronization primitives for thread safety, nor does it rely on dynamic memory.</td>
```

Comment:
202009.00 does not allocate any memory and therefore, has nothing to free.

Reply:
Changed